src/app/dev/DevToys.Tools/Tools/EncodersDecoders/JsonWebToken/JsonWebTokenEncoderGuiTool.cs

```diff
@@ -0,0 +1,605 @@
+﻿using DevToys.Tools.Helpers;
+using DevToys.Tools.Helpers.JsonWebToken;
+using DevToys.Tools.Models;
+using Microsoft.Extensions.Logging;
+
+namespace DevToys.Tools.Tools.EncodersDecoders.JsonWebToken;
+
+internal sealed partial class JsonWebTokenEncoderGuiTool
+{
+    /// <summary>
+    /// Define if the token algorithm
+    /// </summary>
+    private static readonly SettingDefinition<JsonWebTokenAlgorithm> tokenAlgorithmSetting
+        = new(
+            name: $"{nameof(JsonWebTokenEncoderGuiTool)}.{nameof(tokenAlgorithmSetting)}",
+            defaultValue: JsonWebTokenAlgorithm.HS256);
+
+    /// <summary>
+    /// Define if the token has an issuer or not
+    /// </summary>
+    private static readonly SettingDefinition<bool> tokenHasIssuerSetting
+        = new(
+            name: $"{nameof(JsonWebTokenEncoderGuiTool)}.{nameof(tokenHasIssuerSetting)}",
+            defaultValue: false);
+
+    /// <summary>
+    /// Define the issuer list
+    /// </summary>
+    private static readonly SettingDefinition<string> tokenIssuerSetting
```

Comment:
Do we really want to save this string in the settings? We're basically saving a user input on the hard drive by doing this. Settings aren't encrypted. Could this be a risky thing to do?

Same thing for Audience.